ports/rp2/modmachine.c

```diff
@@ -107,10 +107,6 @@ STATIC mp_obj_t machine_freq(size_t n_args, const mp_obj_t *args) {
         if (!set_sys_clock_khz(freq / 1000, false)) {
             mp_raise_ValueError(MP_ERROR_TEXT("cannot change frequency"));
         }
-        #if MICROPY_HW_ENABLE_UART_REPL
-        setup_default_uart();
-        mp_uart_init();
-        #endif
```

Comment:
I think this is still necessary, in order to reinit the UART baudrate based on the new system frequency.

Reply:
That sounds reasonable. At  a first glance, it looked strange. And it should not hurt.